Jip 56 로그인 이후 42 인증 api

tmam444 opened the pull request:
### 개요
 - 로그인 이후 42 intra 인증 api를 구현
 
### 작업 사항
- package.json에 slack library 추가
- 관련된 함수 auth 파일들에 추가

### 변경점
- oauth 로직으로 인해 config 파일이 수정됨.
- .env 파일도 수정해야 합니다. -> slack에 공유하겠음
- app.ts에 passport 유형 추가
- auth routes에 getIntraAuthentication, intraAuthentication 추가
- auth.strategy.ts에 intra 인증 주소 추가
- auth.controller.ts에 getIntraAuthentication, intraAuthentication 함수 추가
- auth.service.ts에 updateAuthenticationUser 함수 추가
- auth.slack.ts에 slack API에 관련한 함수 추가

### 목적
- API 개발

### 스크린샷 (optional)
- 이제 대대적인 Auth 리팩토링을..

naranghae reviewed (approved):
@Expose({ groups: ['] }) 변경 확인했습니다.

tmam444 committed:
JIP-56 config.ts 파일 수정
    - oauth 로직으로 인해 수정하였습니다.
    - .env 파일도 수정해야 합니다.
    - slack에 공유하겠음

tmam444 committed:
JIP-56 42 intra 인증 API 추가
    - app.ts에 passport 유형 추가
    - auth routes에 getIntraAuthentication, intraAuthentication 추가
    - auth.strategy.ts에 intra 인증 주소 추가
    - auth.controller.ts에 getIntraAuthentication, intraAuthentication
      함수 추가
    - auth.service.ts에 updateAuthenticationUser 함수 추가

tmam444 committed:
JIP-56 slack API 추가
    - slack/web-api library package.json에 추가
    - controller, routes, service에 필요한 함수 추가
    - auth.slack.ts에 slack API에 관련한 함수 추가

YeonSeong-Lee commented:
수고하셨습니다.!

jimin52 reviewed (approved):
수고하셨습니다

kth2624 reviewed (approved):
대단합니당
